File: src/main/java/econtroller/actuator/NodeRequest.java
package econtroller.actuator;

import se.sics.kompics.Event;
import se.sics.kompics.address.Address;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-04-23
 *
 */

public class NodeRequest extends Event {

	private Address cloudProviderAddress;

	public NodeRequest(Address cloudProviderAddress) {
		this.cloudProviderAddress = cloudProviderAddress;
	}

	public Address getCloudProviderAddress() {
		return cloudProviderAddress;
	}

}

File: src/main/java/econtroller/actuator/NodeRequest.scala
package econtroller.actuator

import se.sics.kompics.Event
import se.sics.kompics.address.Address

/**
 * @author Amir Moulavi
 * @date 2011-07-13
 *
 */
class NodeRequest(val cloudProviderAddress:Address,
                  val controlInput:Double,
                  val numberOfNodes:Int)
  extends Event